fix for bug 626 - Spotify customization variables

laytonhayes opened the pull request:
Fix for Spotify customization variables **SOUND_SPOTIFY_DISABLE_NORMALISATION** and **SOUND_SPOTIFY_ENABLE_CACHE**. Now the variables can be defined and have no value and work correctly.

Closes: https://github.com/balena-labs-projects/balena-sound/issues/626

maggie0002 commented:
Nice, thanks!

It looks like the script in this container is run by bash though:

`CMD [ "/bin/bash", "/usr/src/start.sh" ]`

So the shebang would be:

`#!/usr/bin/env bash`

With the correct shebang in place, you can then paste the whole script in to this awesome tool to take care of any other syntax: https://www.shellcheck.net

laytonhayes commented:
Thanks for reviewing and pointing me to shellcheck.net, @maggie0002! Great resource. 
I've updated the code based on your comments, rebased, and pushed.

maggie0002 commented:
Looks good to me, thanks. 

Have you been able to test it all? I am not able to on my end, but if you are happy with the fix and to take the wrath of the balenaSound community then I will go ahead and merge.

maggie0002 commented:
Closes: https://github.com/balena-labs-projects/balena-sound/issues/626

lgtm

laytonhayes commented:
@maggie0002 - I've tested it on two devices with all variable permutations and had no issues. I'd rather not face the wrath ;)
I haven't added a fix for #620 but I imagine it's a similar fix. I can have a look after this is merged in. Thanks for reviewing!

maggie0002 commented:
> I'd rather not face the wrath 

Merges only impact new deploys, not old ones, so it's relatively low risk in the sense we can always go back if someone raises an issue. 👍

fgatin commented:
Hello, did this end up fixing #620 or has there been a fix for #620? 
It's quite important to me since I'm trying to run the software on the Pi Zero and without the possibility to disable all these services the Pi is not powerful enough.
